wahanegi/vibereport
# CI-551: [Snyk] Security upgrade web-console from 4.2.0 to 4.2.1

[![CI-551](https://badgen.net/badge/JIRA/CI-551/0052CC)](https://cloverpop-internship.atlassian.net/browse/CI-551) [<img width="16" alt="Powered by Pull Request Badge" src="https://user-images.githubusercontent.com/1393946/111216524-d2bb8e00-85d4-11eb-821b-ed4c00989c02.png">](https://pullrequestbadge.com/?utm_medium=github&utm_source=wahanegi&utm_campaign=badge_info)

<p>This PR was automatically created by Snyk using the credentials of a real user.</p><br /><h3>Snyk has created this PR to fix one or more vulnerable packages in the `rubygems` dependencies of this project.</h3>



#### Changes included in this PR

- Changes to the following files to upgrade the vulnerable dependencies to a fixed version:
    - Gemfile
    - Gemfile.lock



#### Vulnerabilities that will be fixed
##### With an upgrade:
Severity                   | Priority Score (*)                   | Issue                   | Breaking Change                   | Exploit Maturity
:-------------------------:|-------------------------|:-------------------------|:-------------------------|:-------------------------
![medium severity](https://res.cloudinary.com/snyk/image/upload/w_20,h_20/v1561977819/icon/m.png "medium severity")  |  **449/1000**  <br/> **Why?** Has a fix available, CVSS 4.7  | Cross-site Scripting (XSS) <br/>[SNYK-RUBY-ACTIONPACK-5741907](https://snyk.io/vuln/SNYK-RUBY-ACTIONPACK-5741907) |  No  | No Known Exploit 
![medium severity](https://res.cloudinary.com/snyk/image/upload/w_20,h_20/v1561977819/icon/m.png "medium severity")  |  **501/1000**  <br/> **Why?** Recently disclosed, Has a fix available, CVSS 4.3  | Information Exposure <br/>[SNYK-RUBY-ACTIVESUPPORT-5851458](https://snyk.io/vuln/SNYK-RUBY-ACTIVESUPPORT-5851458) |  No  | No Known Exploit 
![medium severity](https://res.cloudinary.com/snyk/image/upload/w_20,h_20/v1561977819/icon/m.png "medium severity")  |  **616/1000**  <br/> **Why?** Proof of Concept exploit, Has a fix available, CVSS 5.9  | Web Cache Poisoning <br/>[SNYK-RUBY-RACK-1061917](https://snyk.io/vuln/SNYK-RUBY-RACK-1061917) |  No  | Proof of Concept 
![medium severity](https://res.cloudinary.com/snyk/image/upload/w_20,h_20/v1561977819/icon/m.png "medium severity")  |  **501/1000**  <br/> **Why?** Recently disclosed, Has a fix available, CVSS 4.3  | Information Exposure <br/>[SNYK-RUBY-RAILTIES-5851410](https://snyk.io/vuln/SNYK-RUBY-RAILTIES-5851410) |  No  | No Known Exploit 

(*) Note that the real score may have changed since the PR was raised.











Check the changes in this PR to ensure they won't cause issues with your project.



------------



**Note:** *You are seeing this because you or someone else with access to this repository has authorized Snyk to open fix PRs.*

For more information:  <img src="https://api.segment.io/v1/pixel/track?data=eyJ3cml0ZUtleSI6InJyWmxZcEdHY2RyTHZsb0lYd0dUcVg4WkFRTnNCOUEwIiwiYW5vbnltb3VzSWQiOiIxZjQ1ZjFjOC1mNGZlLTRiMjItYTU3Yi04OGNkZmUxNGQwNDIiLCJldmVudCI6IlBSIHZpZXdlZCIsInByb3BlcnRpZXMiOnsicHJJZCI6IjFmNDVmMWM4LWY0ZmUtNGIyMi1hNTdiLTg4Y2RmZTE0ZDA0MiJ9fQ==" width="0" height="0"/>
🧐 [View latest project report](https://app.snyk.io/org/rogergraves/project/6998e670-d362-4fcf-a4ad-ddff8c06149d?utm_source&#x3D;github&amp;utm_medium&#x3D;referral&amp;page&#x3D;fix-pr)

🛠 [Adjust project settings](https://app.snyk.io/org/rogergraves/project/6998e670-d362-4fcf-a4ad-ddff8c06149d?utm_source&#x3D;github&amp;utm_medium&#x3D;referral&amp;page&#x3D;fix-pr/settings)

📚 [Read more about Snyk's upgrade and patch logic](https://support.snyk.io/hc/en-us/articles/360003891078-Snyk-patches-to-fix-vulnerabilities)

[//]: # (snyk:metadata:{"prId":"1f45f1c8-f4fe-4b22-a57b-88cdfe14d042","prPublicId":"1f45f1c8-f4fe-4b22-a57b-88cdfe14d042","dependencies":[{"name":"web-console","from":"4.2.0","to":"4.2.1"}],"packageManager":"rubygems","projectPublicId":"6998e670-d362-4fcf-a4ad-ddff8c06149d","projectUrl":"https://app.snyk.io/org/rogergraves/project/6998e670-d362-4fcf-a4ad-ddff8c06149d?utm_source=github&utm_medium=referral&page=fix-pr","type":"auto","patch":[],"vulns":["SNYK-RUBY-ACTIONPACK-5741907","SNYK-RUBY-ACTIVESUPPORT-5851458","SNYK-RUBY-RACK-1061917","SNYK-RUBY-RAILTIES-5851410"],"upgrade":["SNYK-RUBY-ACTIONPACK-5741907","SNYK-RUBY-ACTIVESUPPORT-5851458","SNYK-RUBY-RACK-1061917","SNYK-RUBY-RAILTIES-5851410"],"isBreakingChange":false,"env":"prod","prType":"fix","templateVariants":["updated-fix-title","priorityScore"],"priorityScoreList":[449,501,616,501],"remediationStrategy":"vuln"})

---

**Learn how to fix vulnerabilities with free interactive lessons:**

 🦉 [Cross-site Scripting (XSS)](https://learn.snyk.io/lesson/dom-based-xss/?loc&#x3D;fix-pr)

---
## How to Test
1. Try to execute simple Ruby code in the Heroku console to ensure that it responds to input. For instance:

      `name = "Heroku"`
      `name.upcase `     
      This should return "HEROKU" as the output.

2. Check database calls or other more complex operations if necessary. For instance:

      `AdminUser.create(email: "AdminUser@example.com", password: "Test@2023")`
      After this, ensure that the admin user has been created in the database.
      
      `AdminUser.first`
      This should return the first admin user object from your database.
      
     ` AdminUser.where('created_at > ?', 1.week.ago).count`
      This will return the number of admin users created in the last week.

[CI-551]: https://cloverpop-internship.atlassian.net/browse/CI-551?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
